typegate/src/planner/args.ts

```diff
@@ -816,14 +828,33 @@ class TypeMismatchError extends Error {
   }
 }
 
-const ajv = new Ajv();
+const visitedObject = new Set<string>();
```

Comment:
why is the cache global and not per instance of build?

Reply:
Because ajv is global, `visitedObject` is a subset of all schemas names present in the ajv instance above.

Reply:
Hm, still not clear to me, what happens if a type called "Author" is in a given shape (let's call v1), and then in the same typegraph or another one, the same `wrapped.title` is referencing another shape (v2)? I understand that the type checking might end up wrong?